### PR TITLE
change assert in protection in RecHit producer

### PR DESCRIPTION
**Pull request by VinInn**
somehow this went lost in my last PR
now it runs w/o crash on real data including events (one event) with two very hot modules

### CODE REVIEW

File: RecoLocalTracker/SiPixelRecHits/plugins/SiPixelRecHitHeterogeneous.cc [hunk 1]
```diff
@@ -242,9 +242,9 @@ void SiPixelRecHitHeterogeneous::run(const edm::Handle<SiPixelClusterCollectionN
     std::sort_heap(ind, ind+ngh,[&](auto a, auto b) { return mrp[a]<mrp[b];});
     uint32_t ic=0;
     auto jnd = [&](int k) { return fc+ind[k]; };
-    assert(ngh==DSViter->size());
+    assert(ngh<=DSViter->size());
     for (auto const & clust : *DSViter) {
-      assert(ic<ngh);
+      if(ic>=ngh) break;
```
Comment (fwyzard) [inline RecoLocalTracker/SiPixelRecHits/plugins/SiPixelRecHitHeterogeneous.cc]: what happens when we `break` here ?

Comment (VinInn) [reply RecoLocalTracker/SiPixelRecHits/plugins/SiPixelRecHitHeterogeneous.cc]: that the clusters between ```ngh and DSViter->size()```
are not processed.

This is supposed to be consistent with the cut already implemented in
_RecoLocalTracker/SiPixelRecHits/plugins/gpuPixelRecHits.h_
which reads
```
    if (threadIdx.x==0 && nclus > MaxClusInModule) {
      printf("WARNING: too many clusters %d in Module %d. Only first %d processed\n", nclus,me,MaxClusInModule);
      // zero charge: do not bother to do it in parallel
      for (auto d=MaxClusInModule; d<nclus; ++d) { chargeh[d]=0; detInd[d]=InvId;}
    }
    nclus = std::min(nclus, MaxClusInModule);
```

Comment (VinInn) [reply RecoLocalTracker/SiPixelRecHits/plugins/SiPixelRecHitHeterogeneous.cc]: at the moment the cut in the clusterizer in larger than ```MaxClusInModule``` to account for Clusters below threshold.
One could implement it in _/RecoLocalTracker/SiPixelClusterizer/plugins/gpuClusterChargeCut.h_

**Comment by fwyzard** [comment]
:construction: Validation running at vinavx2.cern.ch:/data/user/fwyzard/patatrack/validation/run.SE2bMtya7E ...

**Pull request merged**